component/discovery/digitalocean/digitalocean.go

```diff
@@ -0,0 +1,72 @@
+package digitalocean
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/common/config"
+	"github.com/grafana/agent/component/discovery"
+	"github.com/prometheus/common/model"
+	prom_discovery "github.com/prometheus/prometheus/discovery/digitalocean"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "discovery.digitalocean",
+		Args:    Arguments{},
+		Exports: discovery.Exports{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+type Arguments struct {
+	RefreshInterval  time.Duration           `river:"refresh_interval,attr,optional"`
+	Port             int                     `river:"port,attr,optional"`
+	HTTPClientConfig config.HTTPClientConfig `river:",squash"`
```

Comment:
I would suggest we duplicate the entire config struct and only expose the fields we think are valid, otherwise we risk undocumented fields showing up in the UI and confusing people.